File: Examples/reflection/Solid.scala
// reflection/Solid.scala
package reflection
import com.atomickotlin.AtomicTest._
import com.atomickotlin.Name

class Solid extends Name
val s = new Solid
s is "Solid"

class Solid2(val size:Int) extends Name {
  override def toString =
    s"${super.toString}($size)"
}

val s2 = new Solid2(47)
s2 is "Solid2(47)"

File: Examples/Reflection/Solid.kt
// Reflection/Solid.kt
import atomictest.eq
import reflection.className

class Solid {
  override fun toString() = className(this)
}

class Solid2(val size: Int) {
  override fun toString() =
    "${className(this)}($size)"
}

fun main(args: Array<String>) {
  Solid() eq "Solid"
  Solid2(47) eq "Solid2(47)"
}
